PublicMapping/districtbuilder
# Add backend to persist locked districts

## Overview

This adds the backend needed to support both #405 and #220 

### Checklist


### Demo

```
$ curl 'http://localhost:3003/api/projects/e1cb7dfc-eafb-4566-bbec-9c325aae58ce' \
  -X 'PATCH' \
  -H 'Connection: keep-alive' \
  -H 'Accept: application/json, text/plain, */*' \
  -H 'Authorization: Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiJiNmJmYjE1OC1kYWZjLTRmZDUtYjkwMi03ZjE1MjNmYmUzZDIiLCJpZCI6ImI2YmZiMTU4LWRhZmMtNGZkNS1iOTAyLTdmMTUyM2ZiZTNkMiIsIm5hbWUiOiJNaWtlIiwiZW1haWwiOiJtbWF1cml6aUBhemF2ZWEuY29tIiwiaXNFbWFpbFZlcmlmaWVkIjpmYWxzZSwiaGFzU2VlblRvdXIiOmZhbHNlLCJpYXQiOjE2MDI4NzY1MDcsImV4cCI6MTYwMzQ4MTMwN30.Px8wtYRqKnYCgH27gt5AOXqxHfZLtMMjYVSW1wBhJG0' \
  -H 'User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/86.0.4240.111 Safari/537.36' \
  -H 'Content-Type: application/json;charset=UTF-8' \
  -H 'Origin: http://localhost:3003' \
  -H 'Sec-Fetch-Site: same-origin' \
  -H 'Sec-Fetch-Mode: cors' \
  -H 'Sec-Fetch-Dest: empty' \
  -H 'Referer: http://localhost:3003/projects/e1cb7dfc-eafb-4566-bbec-9c325aae58ce' \
  -H 'Accept-Language: en-US,en;q=0.9' \
  -H 'Cookie: initialTrafficSource=utmcsr=(direct)|utmcmd=(none)|utmccn=(not set); _ga=GA1.1.1235121353.1601385032; csrftoken=rqCswFuh0w92ZCbQ0PRDWGGd1oBj2oQJidC9ZrjiF2FBYH0P3MsaMYh4Bwd31h6B; sessionid=vjw5g3avd6lqukq9i5jgo74wvgnqbjt4; __utmzzses=1; _gid=GA1.1.287440220.1603374041; _gat_UA-970854-39=1' \
  --data-binary '{"lockedDistricts":[false,false,false,false,false,false,false,false,false,false,true]}' \
  --compressed
{"error":"Bad Request","message":{"lockedDistricts":["Length of array does not match \"number_of_districts\""]}}%  

$ curl 'http://localhost:3003/api/projects/e1cb7dfc-eafb-4566-bbec-9c325aae58ce' \
  -X 'PATCH' \
  -H 'Connection: keep-alive' \
  -H 'Accept: application/json, text/plain, */*' \
  -H 'Authorization: Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiJiNmJmYjE1OC1kYWZjLTRmZDUtYjkwMi03ZjE1MjNmYmUzZDIiLCJpZCI6ImI2YmZiMTU4LWRhZmMtNGZkNS1iOTAyLTdmMTUyM2ZiZTNkMiIsIm5hbWUiOiJNaWtlIiwiZW1haWwiOiJtbWF1cml6aUBhemF2ZWEuY29tIiwiaXNFbWFpbFZlcmlmaWVkIjpmYWxzZSwiaGFzU2VlblRvdXIiOmZhbHNlLCJpYXQiOjE2MDI4NzY1MDcsImV4cCI6MTYwMzQ4MTMwN30.Px8wtYRqKnYCgH27gt5AOXqxHfZLtMMjYVSW1wBhJG0' \
  -H 'User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/86.0.4240.111 Safari/537.36' \
  -H 'Content-Type: application/json;charset=UTF-8' \
  -H 'Origin: http://localhost:3003' \
  -H 'Sec-Fetch-Site: same-origin' \
  -H 'Sec-Fetch-Mode: cors' \
  -H 'Sec-Fetch-Dest: empty' \
  -H 'Referer: http://localhost:3003/projects/e1cb7dfc-eafb-4566-bbec-9c325aae58ce' \
  -H 'Accept-Language: en-US,en;q=0.9' \
  -H 'Cookie: initialTrafficSource=utmcsr=(direct)|utmcmd=(none)|utmccn=(not set); _ga=GA1.1.1235121353.1601385032; csrftoken=rqCswFuh0w92ZCbQ0PRDWGGd1oBj2oQJidC9ZrjiF2FBYH0P3MsaMYh4Bwd31h6B; sessionid=vjw5g3avd6lqukq9i5jgo74wvgnqbjt4; __utmzzses=1; _gid=GA1.1.287440220.1603374041; _gat_UA-970854-39=1' \
  --data-binary '{"lockedDistricts":[false,false,false,false,false,false,false,false,false,false,false,true]}' \
  --compressed
{"id":"e1cb7dfc-eafb-4566-bbec-9c325aae58ce","name":"AZ","numberOfDistricts":12,"districtsDefinition":[1,1,1,1,0,1,1,1,1,1,1,0,1,1,1],"createdDt":"2020-10-16T19:28:55.717Z","advancedEditingEnabled":true,"lockedDistricts":[false,false,false,false,false,false,false,false,false,false,false,true],"regionConfig":{"id":"ca43c843-ba08-4c57-97c1-a64b6a609562","name":"Arizona","countryCode":"US","regionCode":"AZ","s3URI":"s3://global-districtbuilder-dev-us-east-1/regions/US/AZ/2020-10-16T19:25:45.661Z/","version":"2020-10-16T19:25:45.661Z"},"user":{"id":"b6bfb158-dafc-4fd5-b902-7f1523fbe3d2","name":"Mike"}}%
```

## Notes

Also includes a small fix to the migration script (not sure how it was working before)

## Testing Instructions

- The project screen "Accept" button should continue to exercise the project PATCH endpoint as normal
- Attempts to use the project PATCH endpoint to update `lockedDistricts` should work
- Attempts to use the project PATCH endpoint to update a non-updateable field such as `numberOfDistricts` should fail
